### PR TITLE
clean up the implementation around QueueingHintFn 

### PR DESCRIPTION
**Pull request by sanposhiho**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Follow up the comments from @Huang-Wei and @ahg-g: https://github.com/kubernetes/kubernetes/pull/118551#pullrequestreview-1511823059

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

/cc @Huang-Wei 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/scheduler/internal/queue/scheduling_queue.go [hunk 1]
```diff
@@ -866,10 +865,19 @@ func (p *PriorityQueue) requeuePodViaQueueingHint(logger klog.Logger, pInfo *fra
...
+	if added {
 		metrics.SchedulerQueueIncomingPods.WithLabelValues("active", event).Inc()
 		return activeQ
 	}
+	if pInfo.Gated {
+		// In case the pod is gated, the Pod is pushed back to unschedulable Pods pool in addToActiveQ.
+		return unschedulablePods
+	}
+
+	logger.Error(err, "Error adding pod to the active queue, queue this Pod to unschedulable pod pool", "pod", klog.KObj(pod))
```
Comment (Huang-Wei) [inline pkg/scheduler/internal/queue/scheduling_queue.go:880]: it should be added below L870, and wrapped by checking if err is nil, right?

**Comment by ahg-g** [comment]
looks good to me after addressing Wei's last comment related to the error message.

**Comment by sanposhiho** [comment]
Moved the error log place as suggested + squashed. 🙏

**Pull request merged**